src/backend/marsha/core/lti/validator.py

```diff
@@ -0,0 +1,174 @@
+
+        Args:
+            client_key: The client/consumer key.
+            request: The calling request
+
+        Returns:
+            bool: True if the client key is registered and valid
+        """
+        return LTIPassport.objects.filter(
+            oauth_consumer_key=client_key, is_enabled=True
+        ).exists()
+
+    # pylint: disable=too-many-arguments
+    def validate_timestamp_and_nonce(
+        self,
+        client_key: str,
+        timestamp: str,
+        nonce: str,
+        request,
+        request_token=None,
+        access_token=None,
+    ):
+        """Validates that the nonce has not been used before
+
+        Per `Section 3.3`_ of the OAuth 1.0 spec.
+
+        "A nonce is a random string, uniquely generated by the client to allow
+        the server to verify that a request has never been made before and
+        helps prevent replay attacks when requests are made over a non-secure
+        channel.  The nonce value MUST be unique across all requests with the
+        same timestamp, client credentials, and token combinations."
+
+        .. _`Section 3.3`: https://tools.ietf.org/html/rfc5849#section-3.3
+
+        One of the first validation checks that will be made is for the validity
+        of the nonce and timestamp, which are associated with a client key and
+        possibly a token. If invalid then immediately fail the request
+        by returning False. If the nonce/timestamp pair has been used before and
+        you may just have detected a replay attack. Therefore, it is an essential
+        part of OAuth security that you not allow nonce/timestamp reuse.
+        Note that this validation check is done before checking the validity of
+        the client and token.::
+
+        Args:
+            client_key: The client/consumer key.
+            timestamp: The ``oauth_timestamp`` request parameter.
+            nonce: The ``oauth_nonce`` request parameter.
+            request: The calling request
+            request_token: Request token string, if any.
+            access_token: Access token string, if any.
+
+        Returns:
+            bool: True if the timestamp and nonce has not been used before
+        """
+
+        cache_timeout = 3600
+        # Disallow usage of timestamp older than cache_timeout
+        request_timestamp = int(timestamp)
+        if request_timestamp < int(time.time()) - cache_timeout:
+            logger.debug(
+                "Timestamp is too old (ts = %s, consumer_key = %s, nonce = %s)",
+                timestamp,
+                client_key,
+                nonce,
+            )
+            return False
+
+        try:
+            cache = caches[self.LTI_REPLAY_PROTECTION_CACHE]
+        except InvalidCacheBackendError:
+            logger.debug(
+                "Unable to find cache %s, fallback to default cache",
+                self.LTI_REPLAY_PROTECTION_CACHE,
+            )
+            cache = default_cache
```

Comment:
IMO you can directly use the default cache. We only use this one in Marsha and we don't plan to use different one.